Update content type header checking

kenfire opened the pull request:
The verification of the content type now checks if the request content type if contained and not if it's equal.
This allows the have content types such as "application/vnd.api+json; charset=utf-8"

Commit:
Update content type header checking

The verification of the content type now checks if the request content type if contained and not if it's equal.
This allows the have content types such as "application/vnd.api+json; charset=utf-8"

Commit:
Add strings update

derekdowling commented on parser.go:
```diff
@@ -160,7 +161,7 @@ func closeReader(reader io.ReadCloser) {
 
 	reqContentType := headers.Get("Content-Type")
-	if reqContentType != ContentType {
+	if strings.Contains(reqContentType, ContentType) == false {
```
Thoughts on utilizing `strings.HasPrefix` instead here? I assume we always want `application/vnd.api+json` to be the initial part of this header value?

kenfire replied:
I was not sure if the position was defined in the specification. If it's always the first type then yes, it would be better.

derekdowling reviewed (changes requested):
Hey @kenfire thanks for the contribution!

I'm happy to land this once there is a corresponding unit test checking that a request with a `content-type` header of `application/vnd.api+json; charset=utf-8` doesn't get rejected to make sure we don't ever have a future regression around this. Thanks 😄

Commit:
Add Content-Type tests
Add two test cases for the content type
The first one checks if the defautlt content type i.e. application/vnd.api+json; is present
The second checks if the default content type is present when having other field such as charset

CrushedPixel commented:
According to the [jsonapi spec](http://jsonapi.org/format/upcoming/#content-negotiation):

Clients MUST send all JSON API data in request documents with the header Content-Type: `application/vnd.api+json` without any media type parameters.
[...]
Servers MUST respond with a 415 Unsupported Media Type status code if a request specifies the header `Content-Type: application/vnd.api+json` with any media type parameters.


Therefore the current behaviour is fine IMO, and allowing for media type parameters such as `charset=utf-8` would explicitly go against the jsonapi specification.

kenfire commented:
@CrushedPixel is right, therefore this PR should be closed. Might be useful in the future.

derekdowling commented:
Closing per: https://github.com/derekdowling/go-json-spec-handler/pull/39#issuecomment-348914503.
